### PR TITLE
Blacklist nav2 pkgs on Rolling and Iron

### PR DESCRIPTION
**Pull request by Yadunund**
This PR blacklists
* All nav2 pkgs on RHEl 9: See failures https://repo.ros2.org/status_page/ros_iron_rhel.html
* `nav2_bringup` on arm64: https://build.ros2.org/view/Ibin_ujv8_uJv8/job/Ibin_ujv8_uJv8__nav2_bringup__ubuntu_jammy_arm64__binary/

for Rolling and Iron.

cc: @SteveMacenski

### CODE REVIEW

**Review by SteveMacenski — APPROVED** [review]

File: rolling/release-ubuntu-arm64-build.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ package_blacklist:
 - kortex_bringup  # depends on kortex_api and kortex_driver
 - kortex_driver  # depends on kortex_api and uses libKortexApiCpp.a from Kinova
 - mapviz  # Known issues with building on ARM processors. https://github.com/swri-robotics/mapviz/issues/777
+- nav2_bringup # depends on turtlebot3_gazebo which is unavailable
```
Comment (SteveMacenski) [inline rolling/release-ubuntu-arm64-build.yaml]: I don't think this is categorically missing forever and ever to be backlisted. There's work on the gazebo side, but its not unworkable

Comment (Yadunund) [reply rolling/release-ubuntu-arm64-build.yaml]: Understood. In that case I will whitelist it on Rolling but leave it blacklisted on Iron for now.

Comment (SteveMacenski) [reply rolling/release-ubuntu-arm64-build.yaml]: OK. Especially also considering we will eventually migrate completely away from the TB3 and Gazebo classic by EOY. This is an odd transitory state

**Comment by cottsay** [comment]
The reason the nav2 packages are failing on RHEL is due to the dependency on `xtensor` (introduced in ros-planning/navigation2#3350, [rosdep rules](https://github.com/ros/rosdistro/blob/c592c392432cfdd92f6432f2e54a7a0938a82a80/rosdep/base.yaml#L8459-L8465) do not list RHEL, package is currently not available).

Can you update the "reason" in the comment to reflect the missing package as the root cause?

**Comment by Yadunund** [comment]
> The reason the nav2 packages are failing on RHEL is due to the dependency on `xtensor` (introduced in [ros-planning/navigation2#3350](https://github.com/ros-planning/navigation2/pull/3350), [rosdep rules](https://github.com/ros/rosdistro/blob/c592c392432cfdd92f6432f2e54a7a0938a82a80/rosdep/base.yaml#L8459-L8465) do not list RHEL, package is currently not available).
> 
> Can you update the "reason" in the comment to reflect the missing package as the root cause?

ooh I see. Thanks for hunting down the cause. Let me know if the new comment is better.

**Comment by cottsay** [comment]
Thanks, looks good. There is also a linter warning: https://github.com/ros2/ros_buildfarm_config/pull/285/files#diff-9f0c82d1eeac7ed26941b0d693125cb96ccdc63d7f5e220a4445fd7ccc3be1e9

**Comment by SteveMacenski** [comment]
> Can you update the "reason" in the comment to reflect the missing package as the root cause?

I believe there are a few reasons. There's also no OMPL which kills the Smac Planner. I believe there's a couple of others

**Comment by cottsay** [comment]
> I believe there are a few reasons. There's also no OMPL which kills the Smac Planner. I believe there's a couple of others

Having `ompl` blocked here will automatically block downstream packages, so that's not a reason to block these packages here and now.

`xtensor` is the only missing rosdep rule to unblock Bloom in the `navigation2` repository. Once that's resolved, we can remove all of them from this list, at which point `nav2_smac_planner` would still be disabled implicitly due to the `ompl` dependency.

I'm working on getting `xtensor` into EPEL 9.

**Review by cottsay — APPROVED** [review]
It's typically only necessary to block the most upstream packages in this list, as downstream packages would be disabled implicitly, but `navigation2` has a lot of packages and it might be hard to tease apart.

I'm also hoping to get these unblocked in the not-too-distant future, so hopefully this will be short-lived.

**Comment by Yadunund** [comment]
Wow that's great to hear, Scott!

**Pull request merged**